src/ui/public/directives/paginate.js

```diff
@@ -135,6 +136,17 @@ uiModules.get('kibana')
         } else {
           $scope.page = $scope.pages[0];
         }
+
+        //set initial value of paginationCount
+        self.updatePaginationCount(count ? 1 : 0);
+      };
+
+      self.updatePaginationCount = function (number) {
+        $rootScope.paginationCount = 0;
+        for(let i = 0; i < number; i++) {
+          $rootScope.paginationCount = $rootScope.paginationCount + $scope.pages[i].length;
+        }
+        return;
```

Comment:
That's for first fix. It used in https://github.com/elastic/kibana/pull/11440/commits/9db71a3cf5fb8b1ae987b7448049b2b0356cdd4e#diff-c5064ec32d320f6d9cddaa1ed07ed4aeR24
First gif in PR shows that change